src/tests/src/solver/optim-model-filler/test_readLinearExpressionVisitor.cpp

```diff
@@ -105,6 +105,48 @@ struct MockLinearProblemData: Antares::Optimisation::LinearProblemApi::ILinearPr
     }
 };
 
+BOOST_FIXTURE_TEST_CASE(visit_timeSum, MyDummyFixture)
+{ // param  ={0,1,2}
+    // 5 + sum( t-2 .. t+1, param) -->
+    // t = 0 : 5 +param.at(-2) + param.at(-1) + param.at(0) + param.at(1) --> 5 +1 + 2 + 0 + 1 = 9
+    // t = 1 : 5 +param.at(-1) + param.at(0) + param.at(1) + param.at(2) --> 5 + 2+0+1+2 = 10
+    Node* from = create<LiteralNode>(-2.);
+    Node* to = create<LiteralNode>(1.);
+    Node* sum = create<SumNode>(create<LiteralNode>(5.),
+                                create<TimeSumNode>(from, to, create<ParameterNode>("param")));
+    MockLinearProblemData my_data;
+    EvaluationContext evaluation_context(
+      {{build_context_parameter_with("param", "something", ParameterType::TIMESERIE)}},
+      {},
+      my_data);
+    ReadLinearExpressionVisitor visitor(evaluation_context, {0, 2}, componentId);
+    auto linear_expressions = visitor.dispatch(sum).GetLinearExpressions();
```

Comment:
About : 
```c++
auto linear_expressions = visitor.dispatch(sum).GetLinearExpressions();
BOOST_CHECK_EQUAL(linear_expressions.at(0).offset(), 9.);
```
Here we'd like to use `TimeDependentLinearExpression::operator[](int index)` instead (seems more simple) : 
```c++
auto linear_expressions = visitor.dispatch(sum);
BOOST_CHECK_EQUAL(linear_expressions[0].offset(), 9.);
```
but we can't, because this **[] operator** returns a **TimeDependentLinearExpression**.

Reply:
operator[](int index) applied on TimeDependentLinearExpression returns TimeDependentLinearExpression and it's used in ReadLinearExpressionVisitor as well in tests test_linearExpression.cpp::OperatorIndexInvalid and test_linearExpression.cpp::ShiftLinearExpressionsNegativeGreaterThanSize. 
TimeDependentLinearExpression::GetLinearExpressions return map of <timeStep, LinearExpression>

Reply:
and it's true that TimeDependentLinearExpression::GetLinearExpressions().at(Ts) == TimeDependentLinearExpression[Ts]